src/SportsHub.Business/Services/Abstraction/ITeamService.cs

```diff
@@ -19,5 +19,13 @@ public interface ITeamService
         Task<Guid> FindTeamIdBySubcategoryIdAsync(Guid subcategoryId);
 
         Task EditTeamAsync(EditTeamModel editTeamModel);
+
+        IEnumerable<Team> GetTeamsFilteredByLocation(string location, ICollection<Team> teams);
+
+        IEnumerable<Team> GetTeamsFilteredBySubcategoryId(Guid subcategoryId, ICollection<Team> teams);
+
+        IEnumerable<Team> GetTeamsFilteredBySubcategoryIds(IQueryable<Subcategory> subcategories, ICollection<Team> teams);
```

Comment:
Why do you use IQueryable here?
```suggestion
        IEnumerable<Team> GetTeamsFilteredBySubcategoryIds(IQueryable<Subcategory> subcategories, ICollection<Team> teams);
```